test/unstructured_storage.js

```diff
@@ -0,0 +1,61 @@
+const getContract = name => artifacts.require(name)
+
+contract('Constants', accounts => {
+  let app, kernel
+
+  beforeEach(async () => {
+    app = await getContract('AppStubStorage').new()
+    kernel = await getContract('Kernel').new()
+  })
+
+  it('test', async () => {
+    const appId = '0x1234000000000000000000000000000000000000000000000000000000000000'
+    const pinnedCode = '0x1200000000000000000000000000000000005678'
+
+    // set values
+    await app.initialize()
+    await app.setKernelExt(kernel.address)
+    await app.setAppIdExt(appId)
+    await app.setPinnedCodeExt(pinnedCode)
+
+    //checks
+    assert.equal(
+      parseInt(await web3.eth.getStorageAt(app.address, (await app.getInitializationBlockPosition())), 16),
+      (await app.getInitializationBlock.call()).toString(),
```

Comment:
You can check what the block number was after `app.initialize()` and assert equality with the value stored at the storage position.